Problem: No way to represent negative integers

omarkj opened the pull request:
Solution: Provide a way to represent negative
integers (signed) and basic operations on them.

I opened this PR to have further discussions on how we
do numbers in the future within PumpkinDB. There are
some questions to how we:

* Represent signed and unsigned integers;
* Fail when the two types are used together (`10 -5 INT/SUB`);
* Encode signed integers efficiently.

Here I simply append a signing bit at the end of the byte array
generated by the bigint library. There are many much better
ways of doing this, such as using zigzag encoding. The problem
is our use of bigints, instead of sized integers as encoding differs
given how big the integer being signed is.

Another issue is our lack of type tags. We need a way on the binary
level to be able to differentiate between signed and unsigned
integers, the same way we now have different ways to represent
words.

Commit:
Problem: No way to represent negative integers
Solution: Provide a way to represent negative
integers (signed) and basic operations on them.

yrashk commented:
First of all, great start!

>Here I simply append a signing bit at the end of the byte array generated by the bigint library

What would that do to lexicographical ordering, in particular when left-padded?

Commit:
Problem: Lexicographically comparing the signed
integers results in invalid results.

Solution: Move the signified to the first bit.

Commit:
Problem: Documentation for newly added INT
functionality not visible in gitbook.

Solution: Add a link to the newly added
documentation in the `SUMMARY.md` file, which is
used to generate the ToC.

yrashk commented:
Followup suggestion: `UINT->INT` and `INT->UINT` words (they should fail with invalidvalue if they can't be converted). We can do this as a separate PR.
